tonic/src/transport/service/mod.rs

```diff
@@ -22,5 +21,5 @@ pub(crate) use self::io::ServerIo;
 pub(crate) use self::tls::{TlsAcceptor, TlsConnector};
 pub(crate) use self::user_agent::UserAgent;
 
-pub use self::router::Routes;
-pub use self::router::RoutesBuilder;
+/// A deprecated re-export. Please use `tonic::service::{Routes, RoutesBuilder}` directly.
+pub use crate::service::{Routes, RoutesBuilder};
```

Comment:
Should we put `#[deprecated]` on these deprecated re-exports? Why/why not?

Reply:
This is because of the limitation of `#[deprecated]`.

https://github.com/rust-lang/rust/issues/30827.